tools/chpl-language-server/src/chpl-language-server.py

```diff
@@ -1020,6 +1066,31 @@ def _setup_regexes(self):
         # use pat2 first since it is more specific
         self._find_rename_deprecation_regex = re.compile(f"({pat2})|({pat1})")
 
+    def _setup_linter(self, clsConfig: CLSConfig):
+        """
+        Setup the linter, if it is enabled
+        """
+        if not (
+            self.do_linting
+            and chplcheck
+            and chplcheck_lsp
+            and chplcheck_config
+            and chplcheck_driver
+            and chplcheck_rules
+        ):
+            return
+
+        config = chplcheck_config.Config.from_args(clsConfig.args)
+        self.lint_driver = chplcheck_driver.LintDriver(config)
+
+        chplcheck_rules.register_rules(self.lint_driver)
+
+        for p in config.add_rules:
+            chplcheck.load_module(self.lint_driver, os.path.abspath(p))
+
+        self.lint_driver.disable_rules(*config.disabled_rules)
+        self.lint_driver.enable_rules(*config.enabled_rules)
```

Comment:
If you have a `ChplcheckProxy` object, you can make `self.lint_driver` be `self.chplcheck.lint_driver`, and move this code into a method on the proxy. Seems cleaner that way.

Reply:
I would rather keep this code out of a global object